src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -119,73 +119,95 @@ public override Type GetConcreteType(Type parentType)
 
         public override IEnumerable CreateDerivedEnumerableInstance(JsonPropertyInfo collectionPropertyInfo, IList sourceList, string jsonPath, JsonSerializerOptions options)
         {
+            Exception exception = null;
             object instance = collectionPropertyInfo.DeclaredTypeClassInfo.CreateObject();
 
-            if (instance is IList instanceOfIList)
+            try
```

Comment:
I don't think we want this. It's not a good practice to catch all exceptions and re-throw. Do we have any known cases for this yet?

Reply:
> I don't think we want this. It's not a good practice to catch all exceptions and re-throw.

Re-throwing with a predictable exception (and a better message), yes. The problem is that this is effectively calling into "user code", which could throw any exception. How do you document which exception types these APIs could end up throwing? Should you always just wrap these calls in Pokemon catch blocks?

> Do we have any known cases for this yet?

Yes, the original issue (#40237) was about `StringValues` throwing a `NotSupportedException`. Granted, the serializer should've never called that method in the first place (because `IsReadOnly == true`), but it's not hard to imagine a collection throwing from the `Add` method in specific cases.

Reply:
We have other cases of "user code" for example custom converters (and any property setter for that matter) that we don't do a mass catch-all and re-throw.

Since we are not special casing for ReadOnly now we are covered for that case.

Reply:
So you don't want to check `IsReadOnly` and don't `try`/`catch`? Might as well close this PR 😅

Reply:
> We have other cases of "user code" for example custom converters (and any property setter for that matter) that we don't do a mass catch-all and re-throw.

Does that mean every method that touches these converters is documented as possibly throwing any exception type? 🤔